### PR TITLE
Update project UserGuide: assign, cancel

### LINKED ISSUE #38: Update User Guide
Add features to be implemented in v1.2

### PR DESCRIPTION
**Pull request by JinHao-L**
Update User Guide to add commands that support making and deleting appointments.

Added the following segments:
1. Adding an appointment, `assign`
2. Cancel an appointment, `cancel`
3. Archiving data
4. Command summary

#38

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -136,6 +136,36 @@ Examples:
 * `list` followed by `delete 2` deletes the 2nd patient in the address book.
 * `find Betsy` followed by `delete 1` deletes the 1st patient in the results of the `find` command.
 
+### Adding an appointment: `assign`
+
+Assign the specified patient into the specified appointment date and time.
+
+Format: `assign INDEX d/DATE t/TIME`
+
+* Puts the person at the specified INDEX into an appointment time slot.
```
Comment (Avalionnet) [inline docs/UserGuide.md]: Small nit but will it be better to use the word patient here instead?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -136,6 +136,36 @@ Examples:
...
+* The INDEX refers to the index number indicated in the patient list.
+* The IC **must be a positive integer** 1, 2, 3, …​
```
Comment (Avalionnet) [inline docs/UserGuide.md]: Hmm was wondering if IC here will now refer to INDEX instead as we adopted the new command syntax?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -136,6 +136,36 @@ Examples:
...
+* The `DATE` and `TIME` of the appointment must be included.
+* The timeslot indicated by `DATE` and `TIME` must be available.
+* The specified `DATE` and `TIME` must be in the future.
+
+Examples:
+* `assign 1 d/Sunday t/2am` books an appointment at the upcoming Sunday, 2am for the 1st patient in the list.
```
Comment (Avalionnet) [inline docs/UserGuide.md:153]: Perhaps we can consider including another section detailing our acceptance of natural date syntax above to provide users with a clearer understanding of our date syntax.

**Review by Avalionnet — COMMENTED** [review]
LGTM besides the minor comments made 👍🏻 Do check them out :)

File: docs/UserGuide.md (hunk 1 above)
Comment (JinHao-L) [reply docs/UserGuide.md]: Oh yeah. I missed that out. Thanks 👍

File: docs/UserGuide.md (hunk 2 above)
Comment (JinHao-L) [reply docs/UserGuide.md]: Oh I forgot to change it to INDEX. Thanks for spotting it 👍

File: docs/UserGuide.md (hunk 3 above)
Comment (JinHao-L) [reply docs/UserGuide.md:153]: Hmmm, do you think if something like that will be good? [See](https://jinhao-l.github.io/ip/#3-date-time-support)

Comment (Avalionnet) [reply docs/UserGuide.md:153]: Yup! It looks great :)

**Review by Avalionnet — APPROVED** [review]
Looks good after the changes! I'll merge this commit first and we can add in additional info on natural language support in our subsequent iteration.

**Pull request merged**